src/scipp/plotting/resampling_model.py

```diff
@@ -285,11 +285,20 @@ def _resample(self, array):
 
 def _with_edges(array):
     new_array = array.copy(deep=False)
+    if new_array.unit is None:
+        # Workaround required, even if no resampling done
+        new_array.data = array.data.copy()
```

Comment:
```suggestion
        new_array.data = array.data.copy(deep=False)
```
?

Reply:
I think it needs to deep-copy, since we actually change it in the lines below?

Reply:
Do we? I don't see changes to `new_array.data`. And the old code didn't deep copy either.

Reply:
It is changing the unit.